subprojects/core/src/main/java/org/gradle/api/tasks/ant/AntTarget.java

```diff
@@ -33,6 +38,20 @@ public abstract class AntTarget extends ConventionTask {
     private Target target;
     private File baseDir;
 
+    public AntTarget() {
+        getDegradationController().ifPresent(controller -> controller.requireConfigurationCacheDegradation(this,
+            getProject().getProviders().provider(() -> "Task '" + this.getName() + "' of type '" + AntTarget.class.getSimpleName() + "' is not supported with the Configuration Cache")
+        ));
+    }
+
+    private Optional<ConfigurationCacheDegradationController> getDegradationController() {
```

Comment:
💭 I used `ServiceRegistry` to get a class, since in other cases I expose internal type in the API and I get binary compatibility failure